NGitLab.Mock.Tests/GroupsMockTests.cs

```diff
@@ -78,4 +151,192 @@ public void Test_get_subgroups_by_fullpath()
+    }
+
+    [Test]
+    public void Test_get_subgroups_descendants_of_subgroup_by_id()
+    {
+        using var server = CreateGroupHierarchy();
+
+        var client = server.CreateClient("user1");
+        var group = client.Groups.GetSubgroupsAsync(2, new Models.SubgroupQuery { IncludeDescendants = true });
+
+        var expected = new string[] { "tlg/sg1/sg1_1" };
+        Assert.That(group.Select(g => g.FullPath), Is.EquivalentTo(expected));
+    }
+
+    [Test]
+    public async Task Test_page_subgroups_with_descendants_first_page()
+    {
+        using var server = CreateGroupHierarchy();
+
+        var client = server.CreateClient("user1");
+        (var page, var total) = await client.Groups.PageSubgroupsAsync("tlg", new(page: 1, perPage: 3, new Models.SubgroupQuery { IncludeDescendants = true }));
+
+        var expected = new string[] { "tlg/sg1", "tlg/sg1/sg1_1", "tlg/sg2" };
+        Assert.Multiple(() =>
+        {
+            Assert.That(page.Select(g => g.FullPath), Is.EquivalentTo(expected));
+            Assert.That(total, Is.EqualTo(4));
+        });
+    }
+
+    [Test]
+    public async Task Test_page_subgroups_with_descendants_last_page()
+    {
+        using var server = CreateGroupHierarchy();
+
+        var client = server.CreateClient("user1");
+        (var page, var total) = await client.Groups.PageSubgroupsAsync(1, new(page: 2, perPage: 3, new Models.SubgroupQuery { IncludeDescendants = true }));
+
+        var expected = new string[] { "tlg/sg2/sg2_1" };
+        Assert.Multiple(() =>
+        {
+            Assert.That(page.Select(g => g.FullPath), Is.EquivalentTo(expected));
+            Assert.That(total, Is.EqualTo(4));
+        });
+    }
+
+    [Test]
+    public async Task Test_page_subgroups_with_descendants_after_last_page()
+    {
+        using var server = CreateGroupHierarchy();
+
+        var client = server.CreateClient("user1");
+        (var page, var total) = await client.Groups.PageSubgroupsAsync(1, new(page: 100, perPage: 3, new Models.SubgroupQuery { IncludeDescendants = true }));
+
+        Assert.Multiple(() =>
+        {
+            Assert.That(page.Select(g => g.FullPath), Is.Empty);
+            Assert.That(total, Is.EqualTo(4));
+        });
+    }
+
+    [Test]
+    public async Task Test_page_subgroups_with_page_0_returns_page_1()
+    {
+        using var server = CreateGroupHierarchy();
+
+        var client = server.CreateClient("user1");
+        (var page, var total) = await client.Groups.PageSubgroupsAsync("tlg", new(page: 0));
+
+        Assert.That(total, Is.EqualTo(2));
+    }
+
+    [Test]
+    public void Test_page_subgroups_with_invalid_perpage_throws()
+    {
+        using var server = CreateGroupHierarchy();
+        var client = server.CreateClient("user1");
+        Assert.ThrowsAsync<Clients.GitLabBadRequestException>(() => client.Groups.PageSubgroupsAsync(1, new(page: 1, perPage: 0)));
+    }
+
+    private static GitLabServer CreateProjectHierarchy() =>
```

Comment:
Can it be moved at the top of the file instead? Even though I understand it was to keep it close to the related tests, let's keep static fields together.